apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.cs

```diff
@@ -250,26 +269,18 @@ public static void ReleaseToPool(this SpannerClient client, Session session)
                 SessionPoolImpl targetPool = s_poolByClientAndDatabase.GetOrAdd(poolKey,
                                              key => new SessionPoolImpl(key));
 
-
-                SpannerClient evictionClient = poolKey.Client;
-                Session evictionSession = null;
-
                 //Figure out if we want to pool this released session.
-                lock (s_priorityListSync)
+                targetPool.ReleaseSessionToPool(client, session);
+                s_priorityHeap.Add(targetPool);
+                if (CurrentPooledSessions > MaximumPooledSessions)
                 {
-                    targetPool.ReleaseSessionToPool(client, session);
-                    if (CurrentPooledSessions > MaximumPooledSessions)
+                    var evictionPool = s_priorityHeap.GetTop();
+                    var evictionClient = evictionPool.Key.Client;
```

Comment:
hmm  err let me leave the null check in there because I allow the pool size to be managed with few locks meaning a race could result in a null.
good call.